Modularise capture setup script

dporr opened the pull request:
* Modularization of  the whole script so future tweaks, like rollbacks, improved error checking, etc are possible.
* Added a flag --flash-sniffle  for when I just want to flash sniffle but I'm too lazy to copy+paste the commands.
* fixed typo in TB and TBB setup

dporr committed:
Adding pretty-printing helpers.

dporr committed:
modularise env checks

dporr committed:
do checks

dporr committed:
modularised script and add  --flash-sniffle flag.

dporr committed:
fix typo on setup of the aliases for TB and TBB

XenoKovah commented:
What's the correct way to invoke the revised script? I tried doing `source setup_capture_helper_debian-based.sh` and then `do_all` but it just closed the terminal so I couldn't see if there were any error messages.

Edit: It looks like `sudo su` followed by `source setup_capture_helper_debian-based.sh` and `do_all` worked (whereas just `sudo do_all` didn't), but this brings up 2 things / needed improvements:
1) It should have told me that sudo's needed rather than closing the terminal
2) Is there a way to invoke it in a single line like the original "sudo ./setup_capture_helper_debian-based.sh"? I don't want the instructions getting longer (since they're also used in slides)

Other issues:
3) The following error at the end
```
========================================================================================
  Adding helpful command aliases (c, k, TME) to ~/.bashrc
========================================================================================
bash: /root/.cargo/env: No such file or directory
```

Presumably from the environmental differences of the `source /home/$USERNAME/.bashrc` command between a sudo invocation as a user in the original vs using `sudo su` in my test attempt.

4) The final result leaves the user in `/home/user/Blue2thprinting/Sniffle/cc2538-bsl` instead of still in `/home/user/Blue2thprinting/` where they started, which is a problem since their most likely next command may be the setup analysis shell script back in `/home/user/Blue2thprinting/`.

XenoKovah commented:
Also if you could test on a Raspberry Pi ("Zero W" or 4) running bookworm to make sure it still works that'd be appreciated too

dporr commented:
I'll work on your feedback above.

I also have a few RPIs around to test this.

dporr committed:
Merge branch 'darkmentorllc:master' into diego.porras.local

dporr committed:
Return to the base path after performing steps

dporr committed:
Don't close terminal when called with source script.sh

dporr committed:
Run all default steps requiring root

dporr committed:
Error wording

dporr commented:
`1) It should have told me that sudo's needed rather than closing the terminal`
   Fixed by  [74e6045](https://github.com/darkmentorllc/Blue2thprinting/pull/23/commits/74e6045c2180bf8c5bdb6a143388202fe2898dcc) 

`2) Is there a way to invoke it in a single line like the original "sudo ./setup_capture_helper_debian-based.sh"? I don't want the instructions getting longer (since they're also used in slides)`
Fixed by [7607880](https://github.com/darkmentorllc/Blue2thprinting/pull/23/commits/76078805d8ff760ce191261165605cd3d6f2c9df)

Other issues:

`4) The final result leaves the user in /home/user/Blue2thprinting/Sniffle/cc2538-bsl instead of still in /home/user/Blue2thprinting/ where they started, which is a problem since their most likely next command may be the setup analysis shell script back in /home/user/Blue2thprinting/.`
    Fixed by [770d83e](https://github.com/darkmentorllc/Blue2thprinting/pull/23/commits/770d83e90c9b994083eecb1995a01ffc3bd984a0)
   
  I couldn't reproduce the following from my environment. I'm assuming you have rust installed locally and its loaded by default and the SU context switch triggers the error. 
```
    3) The following error at the end
========================================================================================
  Adding helpful command aliases (c, k, TME) to ~/.bashrc
========================================================================================
bash: /root/.cargo/env: No such file or directory

Presumably from the environmental differences of the source /home/$USERNAME/.bashrc command between a sudo invocation as a user in the original vs using sudo su in my test attempt.
```

dporr committed:
Replace hardcoded path in gatttool.c

dporr committed:
Typo

dporr commented:
Tested on rpi 4b running bookworm, see attached log
[rpi_test.log](https://github.com/user-attachments/files/22379521/rpi_test.log)

pi@raspberrypi:~ $ uname -a
Linux raspberrypi 6.12.25+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.12.25-1+rpt1 (2025-04-30) aarch64 GNU/Linux
